### PR TITLE
Implement 'rewrite' command to exclude files from existing snapshots

### LINKED ISSUE #14: Delete Files from Existing Snapshot
In cases of accidential backup of e.g. too large files, I would like to be able to delete specific files or directories (incl. recursion) from existing snapshots

### PR DESCRIPTION
**Pull request by depau**
What does this PR change? What problem does it solve?
-----------------------------------------------------

Since the changes in #2731 haven't seen any progress in the last two years, I thought I'd pick up from where the original author left off and apply the suggested changes.

The current changes include:

- Rebase on top of master + fix API usage
- Apply the trivial review suggestions

This is still WIP.

From the original PR:

> This adds 'rewrite' command that provides a way to remove data from existing snapshots.
> 
> This is some sort of preliminary pull requests. It's actually works here on test repo, but lacks tests, etc. Mostly to discuss things

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------

closes #14
closes #2731 

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added tests for all code changes.
- [ ] I have added documentation for relevant changes (in the manual).
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [ ] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
  - I'm planning to try stuff for the time being, I will rebase appropriately when ready
- [ ] I'm done! This pull request is ready for review.

### CODE REVIEW

**Comment by DRON-666** [comment]
@Depau perhaps you'd be interested in [this fork](https://github.com/DRON-666/restic/commits/cmd-rewrite). I implemented all @aawsome suggestions and added some flags like `--exclude-larger-than` and `--iexclude-file`.
I think other flags like `--set-path`, `--set-time`, `--set-host` should be added as well.

**Comment by ibash** [comment]
Thank you, using this right now!

File: cmd/restic/cmd_rewrite.go [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+		return true, err
+	}
+
+	if opts.Forget {
+		h := restic.Handle{Type: restic.SnapshotFile, Name: sn.ID().String()}
+		if err = repo.Backend().Remove(ctx, h); err != nil {
+			return false, err
+		}
+
+		debug.Log("old snapshot %v forgotten", sn.ID())
```
Comment (ibash) [inline cmd/restic/cmd_rewrite.go:227]: nitpick: IMO it'd be better to make this Verbosef so that the user can see it.

**Comment by Depau** [comment]
> @Depau perhaps you'd be interested in [this fork](https://github.com/DRON-666/restic/commits/cmd-rewrite). I implemented all @aawsome suggestions and added some flags like `--exclude-larger-than` and `--iexclude-file`. I think other flags like `--set-path`, `--set-time`, `--set-host` should be added as well.

Sounds good! You should probably open your own pull request then, I'll close this one.

**Comment by danielsharvey** [comment]
@DRON-666  Will you create a PR for your changes?

**Comment by dionorgua** [comment]
I don't have time right now to work on this. So if anybody want to take care of this pull request, feel free to do this

**Pull request closed without merging**